### PR TITLE
Restructures and integrates collection-level contextual navigation JavaScript

### PR DESCRIPTION
**Pull request by jrgriffiniii**
A proposal for advancing #897.  Documentation should be provided for these changes in order to properly address the related issue.

### CODE REVIEW

File: app/assets/javascripts/arclight/context_navigation.js [hunk 1]
```diff
@@ -308,9 +316,193 @@ class ContextNavigation {
...
+  }
+}
+
+class ArcLightRequestAdapter {
+  /**
+   * Transmit the GET request to the server
+   * This is specific to the Blacklight/ArcLight CatalogController parameters
+   *   for faceting
+   */
+  getData(receivedCallback) {
```
Comment (jrgriffiniii) [inline app/assets/javascripts/arclight/context_navigation.js]: I really would prefer to an approach using `async`/`await` for returning the AJAX responses from this method, but I believe that this cannot be done without addressing Babel transpiling issues (please see https://github.com/projectblacklight/arclight/issues/869).

File: app/assets/javascripts/arclight/context_navigation.js [hunk 2]
```diff
@@ -308,9 +316,193 @@ class ContextNavigation {
...
+}
+
+class CollectionElement {
+  /**
+   * Update the view DOM with the AJAX response from the server
+   */
+  updateView($doc) {
+    let siblings = $doc.find('#documents');
+    // *Important:* this must be called before CollectSiblingTree modifies the
+    // DOM
```
Comment (jrgriffiniii) [inline app/assets/javascripts/arclight/context_navigation.js]: I need to address this before this draft is opened as a Pull Request

File: app/assets/javascripts/arclight/context_navigation.js [hunk 3]
```diff
@@ -308,9 +316,193 @@ class ContextNavigation {
...
+   */
+  getData() {
+    // Add a placeholder so flashes of text are not as significant
+    const $placeholder = new Placeholder();
+    this.$element.after($placeholder);
+
+    const updateViewCallback = (responseBody) => {
+      const $responseDocument = $(responseBody);
+
+      this.updateView($responseDocument);
```
Comment (jrgriffiniii) [inline app/assets/javascripts/arclight/context_navigation.js]: Please see https://github.com/projectblacklight/arclight/pull/934/files#diff-420333906b23e241c8ec41ea7665f017R373

File: app/assets/javascripts/arclight/arclight.js [hunk 4]
```diff
@@ -1,5 +1,4 @@
 //= require arclight/collection_scrollspy
-//= require arclight/collection_navigation
```
Comment (jrgriffiniii) [inline app/assets/javascripts/arclight/arclight.js]: This needs to be restored.

File: app/assets/javascripts/arclight/context_navigation.js [hunk 5]
```diff
@@ -308,9 +316,256 @@ class ContextNavigation {
...
+  $('.al-contents').each(function (i, element) {
+    const $element = $(element);
+
+    // Construct the root node
+    new CollectionElement($element);
+
+    // This is propagated when each element has a child
+    $element.on('navigation.contains.elements', () => {
+      const $target = $(event.target);
+      new CollectionContentTab($target, $toEnableElement, $toSrOnlyElement);
```
Comment (jrgriffiniii) [inline app/assets/javascripts/arclight/context_navigation.js]: This needs to be migrated back into `arclight/collection_navigation`

File: app/assets/javascripts/arclight/arclight.js (hunk 4 above)
Comment (jrgriffiniii) [reply app/assets/javascripts/arclight/arclight.js]: I've restored this with https://github.com/projectblacklight/arclight/pull/934/commits/dbb6821eb3e4e100beec5ac099591b1263bf06ad

File: .eslintrc [hunk 6]
```diff
@@ -11,5 +11,14 @@
   "rules": {
     "no-param-reassign": [2, { "props": false }],
     "func-names": ["error", "never"]
-  }
+  },
+  "overrides": [
+    {
+      "excludedFiles": ["app/assets/javascripts/arclight/context_navigation.js"],
```
Comment (jrgriffiniii) [inline .eslintrc]: Following https://eslint.org/docs/user-guide/configuring#disabling-rules-only-for-a-group-of-files, I couldn't manage to get this passing without violations.  I will attempt adjusting this.

Comment (mejackreed) [reply .eslintrc]: I think thats fine for now.. Inline block disables could also work? (maybe there was an issue with that?)

**Comment by mejackreed** [comment]
With regard to dbb6821. Can we just get `collection_navigation.js` back into its preexisting state for this PR, and move that refactor to a different branch for separate review please?

File: .eslintrc (hunk 6 above)
Comment (jrgriffiniii) [reply .eslintrc]: This doesn't seem to be effective (please see https://travis-ci.org/projectblacklight/arclight/jobs/596661383), I'll try inline block disables.

**Comment by jrgriffiniii** [comment]
https://travis-ci.org/projectblacklight/arclight/builds/596681514 is passing, so perhaps https://travis-ci.org/projectblacklight/arclight/builds/596681511 is failing due to flapping tests

**Pull request merged**